server/container/guice/blob-export-guice/src/main/java/org/apache/james/modules/BlobExportImplChoice.java

```diff
@@ -0,0 +1,106 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.modules;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.apache.commons.configuration.Configuration;
+
+import com.github.steveash.guavate.Guavate;
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+class BlobExportImplChoice {
+
+    enum BlobExportImplName {
+        LOCAL_FILE("localFile");
+
+        private static Optional<BlobExportImplName> from(String implNameString) {
+            Preconditions.checkNotNull(implNameString);
+
+            return Stream.of(values())
+                .filter(impl -> impl.name.equals(implNameString))
```

Comment:
ignoring case  + stripping white spaces can be admin friendly

Reply:
stripping spaces should be handle before that method.
and I'm against ignoring case